src/Npgsql/SqlQueryParser.cs

```diff
@@ -84,12 +78,12 @@ class SqlQueryParser
                 case '"':
                     goto DoubleQuoted;
                 case ':':
```

Comment:
```suggestion
                case ':' when !positionalParameters: 
```

Reply:
Thought about this, but it seemed better to leave the switch to be simple, without any `when` clauses - at least in theory a jump table may get generated (but the moment you put `when` that's not possible).